src/algebra/order/hom/monoid.lean

```diff
@@ -171,6 +171,20 @@ lemma map_nonpos (ha : a ≤ 0) : f a ≤ 0 := by { rw ←map_zero f, exact orde
 
 end ordered_add_comm_monoid
 
+section ordered_add_comm_group
+variables [ordered_add_comm_group α] [ordered_add_comm_monoid β] [add_monoid_hom_class F α β]
+  {f : F}
+
+lemma monotone_iff_map_nonneg : monotone (f : α → β) ↔ ∀ a, 0 ≤ a → 0 ≤ f a :=
+⟨λ h a, by { rw ←map_zero f, apply h }, λ h a b hl,
+  by { rw [←sub_add_cancel b a, map_add f], exact le_add_of_nonneg_left (h _ $ sub_nonneg.2 hl) }⟩
+
+lemma monotone_iff_map_nonpos : monotone (f : α → β) ↔ ∀ a ≤ 0, f a ≤ 0 :=
+⟨λ h a, by { rw ←map_zero f, apply h }, λ h a b hl,
+  by { rw [←sub_add_cancel a b, map_add f], exact add_le_of_nonpos_left (h _ $ sub_nonpos.2 hl) }⟩
```

Comment:
I think you can get that one from the previous by using `αᵒᵈ`, `βᵒᵈ`, or from the one I suggested by using `βᵒᵈ`.

Reply:
Thanks for the suggestions!
The simplest proof I can find via order_dual is the following; let me know if you have a better one!
```
lemma monotone_iff_map_nonpos : monotone (f : α → β) ↔ ∀ a ≤ 0, f a ≤ 0 :=
let f' : αᵒᵈ →+ βᵒᵈ := (f : α →+ β) in 
by { rw [monotone, forall_swap], exact monotone_iff_map_nonneg f' }
```

Reply:
Update: I'm able to golf it to `forall_swap.trans $ monotone_iff_map_nonneg (id (f : α →+ β) :  αᵒᵈ →+ βᵒᵈ)` which seems hard to beat.

Reply:
You should use `monotone_on_comp_of_dual_iff` and friends instead of explicitly calling `forall.swap`. But mostly you don't need to swap the variables when you only turn `β` around (because the swap is for the `a < b` bit, not the `f a < f b` one).

Reply:
Okay I'll replace the proof with
```
lemma monotone_iff_map_nonpos : monotone (f : α → β) ↔ ∀ a ≤ 0, f a ≤ 0 :=
by { rw [←antitone_comp_of_dual_iff, ←monotone_to_dual_comp_iff], exact monotone_iff_map_nonneg _ } 
```
and base other proofs on the same pattern. This is slightly longer but doesn't break API barrier and abuse defeq and is probably easier to maintain.